src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -1,161 +1,132 @@
 using Ocelot.Responses;
 
-namespace Ocelot.DownstreamRouteFinder.UrlMatcher
+namespace Ocelot.DownstreamRouteFinder.UrlMatcher;
+
+public class UrlPathPlaceholderNameAndValueFinder : IPlaceholderNameAndValueFinder
 {
-    public class UrlPathPlaceholderNameAndValueFinder : IPlaceholderNameAndValueFinder
+    private const string PlaceHolderLeft = "{";
+    private const string PlaceHolderRight = "}";
+
+    /// <summary>
+    /// Finds the placeholders in the request path and query and returns their matching values.
+    /// We might encounter the following scenarios:
+    /// - The path template contains a catch-all query parameter. If so, we return the catch-all placeholder with an empty value.
+    /// - The path template contains a catch-all path parameter. If so, we return the catch-all placeholder with an empty value.
+    /// - The path template contains placeholders. We return the placeholders with their matching values.
+    /// </summary>
+    /// <param name="path">The request path.</param>
+    /// <param name="query">The query parameters.</param>
+    /// <param name="pathTemplate">The request path template.</param>
+    /// <returns>The list of the placeholders with their matching values.</returns>
+    public Response<List<PlaceholderNameAndValue>> Find(string path, string query, string pathTemplate)
     {
-        public Response<List<PlaceholderNameAndValue>> Find(string path, string query, string pathTemplate)
+        (bool isCatchAllQuery, string catchAllQueryPlaceholder) = IsCatchAllQuery(pathTemplate);
```

Comment:
Styling and convention issue of using `Tuple` in a method which should simply return `bool` only.
```suggestion
        bool isCatchAllQuery = IsCatchAllQuery(pathTemplate, out string catchAllQueryPlaceholder);
```

Reply:
Not an issue